bugbug/bug_features.py

```diff
@@ -808,10 +807,21 @@ def apply_transform(bug):
                     "comments": " ".join(comments),
                 }
 
-        for bug in bugs():
-            if isinstance(bug, dict):
-                results.append(apply_transform(bug))
-            elif isinstance(bug, tuple):
+        if not is_couple:
+            if self.rollback:
+                with Pool() as p:
+                    return pd.DataFrame(
+                        apply_transform(bug)
+                        for bug in p.imap(
+                            partial(bug_snapshot.rollback, when=self.rollback_when),
+                            bugs_iter,
+                            chunksize=1024,
+                        )
+                    )
```

Comment:
This is kind of deeply nested. We could extract it in a function (e.g., `apply_rollback()`) that returns a generator of rolled-back bugs. With that, if need to rollback, we just replace the bugs iterator:
```py
if self.rollback:
    bugs_iter = apply_rollback(bugs_iter)
return pd.DataFrame(apply_transform(bug) for bug in bugs_iter)
```

Reply:
I've improved this a bit in 71e7552.